### PR TITLE
added patient create button

### PR DESCRIPTION
**Pull request by shachar058**
Added a "create patient" button to the patient search screen.

(I'm not sure if I put the css in the right files.)

below added by bradymiller:
Up for grabs demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Beta_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by bradymiller** [comment]
hi @zbig01 , was hoping you could check this over to ensure styling changes done correctly and to ensure changes cover the color styles. thanks, -brady

**Comment by zbig01** [comment]
Will do

File: interface/main/finder/dynamic_finder.php [hunk 1]
```diff
@@ -160,12 +160,15 @@ function persistCriteria(el, e){
...
-            <div class="col-sm-12">
+            <div class="col-sm-10">
                 <h2>
                 <?php echo xlt('Patient Finder') ?> <i id="exp_cont_icon" class="fa <?php echo attr($expand_icon_class);?> oe-superscript-small expand_contract"
                 title="<?php echo attr($expand_title); ?>" aria-hidden="true"></i> <i id="show_hide" class="fa fa-search-plus fa-2x small" title="<?php echo xla('Click to show advanced search'); ?>"></i>
                 </h2>
             </div>
+            <div class="col-sm-2">
+                <button id="create_patient_btn" onclick="top.restoreSession();location='<?php echo $web_root ?>/interface/new/new.php'"><?php echo xlt('Create Patient'); ?></button>
+            </div>
```
Comment (bradymiller) [inline interface/main/finder/dynamic_finder.php]: Should only show this if have following acl:
```
acl_check('patients', 'demo', '', array('write','addonly'))
```

**Comment by zbig01** [comment]
Here are some of my observations regarding the 'Create New Patient' button in the Patient Finder page
    - Placement on top right/left corner of the page breaks several page standardizations that we are trying to implement. 
    - The `OemrUI class` was written in an effort to standardize the way a page is displayed with the 'Heading' line having some action icons and the Help icon
    - The Help icon is placed on the top right/left corner to ensure a consistent place to look for a context sensitive help 
    - Action buttons that Add/Edit etc should be placed below the 'Heading' line usually on the top left/right corner as the case may be.
    - Do remember the top right/left corner on the 'Heading' line is designed to display a help icon even if one is not implemented for that page at that time
    - Relocating the Create New Patient button on the top left results in a scenario where identical buttons result in 2 slightly different actions, i.e Patient Finder > Create New Patient redirects to the Patient/Client > New/Search Page that has its own Create New Patient button that does what it is supposed to do.


patient_finder_01


patient_finder_02

**Comment by shachar058** [comment]
@zbig01 I hear what you are saying, and I will address each issue:
- I do not think a help button is something that every view needs to leave space for, because a view does not generally have one. 
- There are indeed several views that use the OemrUI class (it's used in a little less than 20 files if I counted correctly) but most of them don't, including this one. I think refactoring to use this class is a job in itself, and should not prevent from adding features to the current code.
- About both buttons having the same name: Although I think the context of the views makes it clear what they do, we could rename to "New/Search" like "Patient/Client" -> "New/Search" in the menu to make it clearer.

@bradymiller, what are your thoughts?

**Comment by shachar058** [comment]
@zbig01 thank you for your input, and I will address each issue:

I do not think a help button is something that every view needs to leave space for, because a view does not generally have one.
There are indeed several views that use the OemrUI class (it's used in a little less than 20 files if I counted correctly) but most of them don't, including this one. I think refactoring to use this class is a job in itself, and should not prevent from adding features to the current code.
About both buttons having the same name: Although I think the context of the views makes it clear what they do, we could rename to "New/Search" like "Patient/Client" -> "New/Search" in the menu to make it clearer.
@bradymiller, what are your thoughts?

**Comment by sjpadgett** [comment]
@zbig01 While you're in a magnanimous mood, do ya think you could help me with setting up a help using your help system for the improved Eligibility feature? There'd be three place. Visit->Eligibility, Visit->Eligibility Response and demographics Eligibility tab.
Resume normal scheduled programming...

**Comment by zbig01** [comment]
@sjpadgett 
That should not be a problem. I will however have to restyle those pages. I am trying to figure out where the demographics Eligibility tab is.
Pl provide the text that you want included in the help files.

**Comment by sjpadgett** [comment]
yep, let me open an issue we can work from and get out of this pr. The eligibility tab is the tab next to insurance in Dashboard and I will need to give you a populated table for testing. Thanks for the help and i'll open issue now

**Comment by bradymiller** [comment]
Up for grabs demo for this PR is here: https://www.open-emr.org/wiki/index.php/Development_Demo#Beta_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
Noted following issue on testing since the search/new gui is not supposed to be in the same frame as the search gui. So, if you have a user opened up, and then use that new button in the search gui to create a new patient, then open door to having 2 different patient summary screens:
image
image

**Comment by bradymiller** [comment]
Also, it does not look like styling is in the color styles. I think need to add to interface/themes/color_base.scss

**Comment by zbig01** [comment]
I suggest moving the Create Patient button to be below and in line with Patient Finder heading in order to achieve uniformity in styling :)

**Comment by shachar058** [comment]
Indeed the button is supposed to be inline with the Patient Finder, but apparently I only put the styling in the light theme. Like Brady said, I need to move it to the color_base.scss file.

The feature you suggested is interesting, but I'm not sure it's a good idea to automatically redirect. There can be situations where we want to check if several patients exist, or where the patient does exist but we searched using the wrong information and we need to search again.

File: interface/themes/style_light.scss [hunk 2]
```diff
@@ -308,6 +308,7 @@ div.category-display {
 .chevron_color {
     color: #2672ec;
 }
+
```
Comment (bradymiller) [inline interface/themes/style_light.scss]: you also need it in the style_light.scss script

File: interface/main/finder/dynamic_finder.php (hunk 1 above)
Comment (bradymiller) [reply interface/main/finder/dynamic_finder.php]: Also need to address the above comment.

File: interface/themes/style_light.scss (hunk 2 above)
Comment (shachar058) [reply interface/themes/style_light.scss]: done . I'm pretty confused about this new styling mechanism - what goes where

File: interface/main/finder/dynamic_finder.php (hunk 1 above)
Comment (shachar058) [reply interface/main/finder/dynamic_finder.php]: done, still need to work on issue with 2 patients being open at the same time

**Comment by shachar058** [comment]
@bradymiller I think I fixed all the issues you brought up, ready for next step :)

**Comment by bradymiller** [comment]
hi @shachar058 , Testing well and appears we have come to a good compromise between you and @zbig01 . So bringing this into the codebase. thanks! -brady

**Pull request merged**